File: winery_backend/src/main/java/bg/tu/varna/si/dto/WineTypeMaxLitersDTO.java
package bg.tu.varna.si.dto;

import java.util.List;

public class WineTypeMaxLitersDTO {
    public Long wineTypeId;
    public Double maxLiters;
    public String limitingVarietyName;
    public List<VarietyLimitDTO> limits;
    public static class VarietyLimitDTO {
        public Long varietyId;
        public String varietyName;
        public Double availableKg;
        public Double kgPerLiter;
        public Double maxLitersForVariety;
    }
}

File: winery_app/app/src/main/java/bg/tu/varna/si/winery/dto/WineTypeMaxLitersDto.kt
package bg.tu.varna.si.winery.dto

import com.google.gson.annotations.SerializedName

data class WineTypeMaxLitersDto(
    @SerializedName("wineTypeId") val wineTypeId: Long,
    @SerializedName("maxLiters") val maxLiters: Double,
    @SerializedName("limitingVarietyName") val limitingVarietyName: String?,
    @SerializedName("limits") val limits: List<VarietyLimitDto> = emptyList()
) {
    data class VarietyLimitDto(
        @SerializedName("varietyId") val varietyId: Long,
        @SerializedName("varietyName") val varietyName: String,
        @SerializedName("availableKg") val availableKg: Double,
        @SerializedName("kgPerLiter") val kgPerLiter: Double,
        @SerializedName("maxLitersForVariety") val maxLitersForVariety: Double
    )
}